development/benchmark-pr-comment.ts

```diff
@@ -0,0 +1,505 @@
+      const p95Value =
+        pageSummary.p95[metric as keyof Omit<BenchmarkMetrics, 'memoryUsage'>];
+      const p99Value =
+        pageSummary.p99[metric as keyof Omit<BenchmarkMetrics, 'memoryUsage'>];
+
+      comment += `| ${metric} | ${formatTime(currentValues.mean)} | ${formatTime(currentValues.stdDev)} | ${formatTime(minValue || 0)} | ${formatTime(maxValue || 0)} | ${formatTime(p95Value || 0)} | ${formatTime(p99Value || 0)} |\n`;
+    }
+
+    comment += `\n</details>\n\n`;
+  }
+
+  // Add warning section if there are significant increases
+  if (significantIncreases.length > 0) {
+    comment += `## ⚠️ Performance Warning\n\n`;
+    comment += `**🚨 Significant performance regression detected!**\n\n`;
+    comment += `The following metrics have increased by a significant amount compared to the reference commit. This should be investigated before proceeding with this PR:\n\n`;
+
+    for (const increase of significantIncreases) {
+      const currentFormatted = formatTime(increase.current);
+      const referenceFormatted = formatTime(increase.reference);
+      comment += `- **${increase.page} - ${increase.metric}**: ${currentFormatted} vs ${referenceFormatted} (**+${increase.percentIncrease.toFixed(1)}%**)\n`;
+    }
+
+    comment += `\n`;
+  }
+
+  comment += `---\n\n`;
+  comment += `*Results generated automatically by MetaMask CI*`;
+
+  return comment;
+}
+
+/**
+ * Main function that orchestrates the benchmark comment posting process.
+ * Reads benchmark results from file, generates a formatted comment, and posts it to the pull request.
+ *
+ * Required environment variables:
+ * - PR_COMMENT_TOKEN: GitHub token with permission to comment on PRs
+ * - OWNER: Repository owner (ex.: "MetaMask")
+ * - REPOSITORY: Repository name (ex.: "metamask-extension")
+ * - PR_NUMBER: Pull request number to comment on
+ *
+ * @throws {Error} When GitHub API request fails or required environment variables are missing
+ */
+async function main(): Promise<void> {
+  const { PR_COMMENT_TOKEN, OWNER, REPOSITORY, PR_NUMBER } =
+    process.env as Record<string, string>;
+
+  if (!PR_NUMBER) {
+    console.warn('No pull request detected, skipping benchmark comment');
+    return;
+  }
+
+  if (!PR_COMMENT_TOKEN) {
+    console.warn('PR_COMMENT_TOKEN not provided, skipping benchmark comment');
+    return;
+  }
+
+  const filePath = path.join(
+    process.cwd(),
+    'test-artifacts/benchmarks/benchmark-results.json',
+  );
+
+  let benchmarkData: BenchmarkOutput | null = null;
+
+  try {
+    const fileContent = await fs.readFile(filePath, 'utf8');
+    benchmarkData = JSON.parse(fileContent) as BenchmarkOutput;
+    console.log(`Found benchmark results at: ${filePath}`);
+  } catch (error) {
+    console.log(`Could not read benchmark results from: ${filePath}`);
+  }
+
+  if (!benchmarkData) {
+    console.warn('No benchmark results found, skipping comment');
+    return;
+  }
+
+  const referenceData = await fetchLatestMainBenchmarkData();
```

Comment:
Should this reference data be against the mean of the (means of the) last N commits on main instead of just the last commit? This would give us more of a recent historical snapshot to compare against each time this runs?